src/components/product.js

```diff
@@ -1,13 +1,110 @@
-import { useNavigate } from "react-router-dom";
+import { useNavigate, useLocation } from "react-router-dom";
 import { useCartContext } from "../context/cart-context";
+import { useAuth } from "../context/auth-context";
+
+import {
+  updateCart,
+  addToCart,
+  removeFromWishlist,
+  addToWishlist,
+} from "../utilities";
 
 function Product(props) {
   const { cartState, cartDispatch } = useCartContext();
-
+  const { authToken } = useAuth();
   const { image, title, condition, brand, price, rating, _id } = props.product;
+  const location = useLocation();
 
   const navigate = useNavigate();
 
+  function cartAction() {
+    if (authToken) {
+      const isItemAlreadyInCart = cartState.cart.find(
+        (item) => item._id === _id
+      );
+      if (isItemAlreadyInCart) {
+        (async () => {
+          try {
+            const updatedCart = await updateCart(
+              authToken,
+              props.product._id,
+              "increment"
+            );
+            console.log(updatedCart);
+            cartDispatch({
+              type: "ADD_T0_CART",
+              payload: updatedCart.data.cart,
+            });
+          } catch (error) {
+            console.log(error);
+          }
+        })();
+      } else {
+        (async () => {
+          try {
+            const updatedCart = await addToCart(authToken, props.product);
+            console.log(updatedCart);
```

Comment:
remove console.log

Reply:
Thanks for pointing out, removed console.log

Reply:
Thanks for pointing out, removed console log